packages/vite/src/node/plugins/define.ts

```diff
@@ -9,7 +9,6 @@ import { isHTMLRequest } from './html'
 const nonJsRe = /\.json(?:$|\?)/
 const isNonJsRequest = (request: string): boolean => nonJsRe.test(request)
 const importMetaEnvMarker = '__vite_import_meta_env__'
-const bareImportMetaEnvRe = new RegExp(`${importMetaEnvMarker}(?!\\.)\\b`)
 const importMetaEnvKeyRe = new RegExp(`${importMetaEnvMarker}\\..+?\\b`, 'g')
```

Comment:
Thank for the quick follow up :pray:
The approach seems right, but I think `importMetaEnvKeyRe` also needs to use an incremented marker. Otherwise `undefined` replacement won't kick in for the code like this:

```ts
        console.log(__vite_import_meta_env__)
        console.log(import.meta.env.UNDEFINED)                           
```

Reply:
I'm sorry to forget this. Now it works.